### PR TITLE
Use a non-empty DEFAULT_DATA_DIR

### PR DESCRIPTION
**Pull request by brson**
This fixes a problem where, if one is using the default, then on
startup, the check_data_dir sanity check fails, printing a warning to
the console that would be easy to ignore.

I gather this doesn't cause problems elsewhere because other code is
calling `join` to create subpaths.

It seems to me that having an empty default is error-prone (as here), so the fix should be to not willingly set it to "" (additional startup checks that no paths are "" would be even better).

I put a special case into the validation code to account for the change from "" to "./" on upgrade. I have tested it manually.

## What have you changed? (mandatory)

Set `DEFAULT_DATA_DIR` to "./".

## What are the type of the changes? (mandatory)

- Bug fix (change which fixes an issue)

## How has this PR been tested? (mandatory)

Manually.

## Refer to a related PR or issue link (optional)

Fixes https://github.com/tikv/tikv/issues/4392

### CODE REVIEW

**Comment by siddontang** [comment]
Good catch, mostly we pass a data dir explicitly so not meeting this problem.

PTAL @zhangjinpeng1987

File: src/config.rs [hunk 1]
```diff
@@ -1330,11 +1330,17 @@ impl TiKvConfig {
         }
 
         if last_cfg.storage.data_dir != self.storage.data_dir {
-            return Err(format!(
-                "storage data dir have been changed, former data dir is {}, \
-                 current data dir is {}, please check if it is expected.",
-                last_cfg.storage.data_dir, self.storage.data_dir
-            ));
+            // In v3 the default value of storage.data-dir changed
```
Comment (siddontang) [inline src/config.rs]: what does V3 mean here?

Comment (Hoverbear) [reply src/config.rs]: Configuration version 3 (Do we document how we support old/new configs?)

File: src/config.rs [hunk 2]
```diff
@@ -1330,11 +1330,17 @@ impl TiKvConfig {
...
+            // from "" to "./"
+            let using_default_after_upgrade =
+                last_cfg.storage.data_dir == "" && self.storage.data_dir == "./";
```
Comment (AndreMouche) [inline src/config.rs]: Could we change it like  `if data_dir.is_empty() or data_dir == DEFAULT_DATA_DIR`? I think it's more clear for the reviewers

File: src/config.rs (hunk 1 above)
Comment (siddontang) [reply src/config.rs]: I think we don't have

Comment (brson) [reply src/config.rs]: I meant TiKV version 3, the next release. I will clarify in the code.

File: src/config.rs (hunk 2 above)
Comment (brson) [reply src/config.rs]: Yes, nice suggestion.

**Comment by brson** [comment]
I've pushed a commit that addresses @siddontang and @AndreMouche's comments.

File: src/config.rs [hunk 3]
```diff
@@ -1330,11 +1331,18 @@ impl TiKvConfig {
...
 
         if last_cfg.storage.data_dir != self.storage.data_dir {
-            return Err(format!(
-                "storage data dir have been changed, former data dir is {}, \
-                 current data dir is {}, please check if it is expected.",
-                last_cfg.storage.data_dir, self.storage.data_dir
-            ));
+            // In tikv 3.0 the default value of storage.data-dir changed
+            // from "" to "./"
+            let using_default_after_upgrade =
```
Comment (siddontang) [inline src/config.rs]: can we add a test that using empty before?

Comment (brson) [reply src/config.rs]: I may not understand your question, but on line 1337, `last_cfg.storage.data_dir.is_empty()` is testing that their previous configuration used the empty string.

**Comment by Hoverbear** [comment]
\run-integration-tests
